### ISSUE TITLE
Unable to set the Volume before starting to play an audio file

### ISSUE THREAD

**Issue by lucasloss** [ZeBobo5/Vlc.DotNet#130]
I noticed that every time I start the application and play an audio, the VLC sounds like it is 100% (or something around that).
I tried setting the volume before playing the audio, but it didn't work.
If I debug the code, I see that the volume is always set to -1.
For instance, if I execute the following lines of code, when I debug it, the volume is still -1:

``` C#
myVlcControl.Play(new FileInfo(FileName));
myVlcControl.Audio.Volume = 40; 
```

The funny thing is (OK, it's not the funny), when the audio is already playing and I change the volume value on my NumericUpDown component, the Volume is successfully changed to the select value. The code below is the event where this happens:

``` C#
private void numericUpDown1_ValueChanged(object sender, EventArgs e)
{
    myVlcControl.Audio.Volume = (int)numericUpDown1.Value;
}
```

I've been trying to solve this problem for two days now. Unfortunately, my coding skills are not even close the people behind this project. That being said:
- Does anyone have the same problem?
- Does anyone know how to solve it?

Any help is appreciated! =)

By the way: I would also like to say that this Vlc.DotNet is an amazing piece of software. It is everything I need to continue one project I'm currently working on, where I need to be able to play most of the existing audio formats (and that's why I chose to use VLC).

**Comment by higankanshi** [comment]
I have found the problem.
 You should use LibVlc 2.2.0(or later).

Vlc.DotNet is using LibVlc 2.1.5

**Comment by lucasloss** [comment]
@higankanshi,
Thanks for helping on such short notice. You're right. Using the LibVlc 2.2.0 I'm able to set the Volume before playing. 

Unfortunately, for some reason, setting the volume before playing the audio only works on the first time the application is opened. After stopping the audio, changing the volume, and playing it again, the volume doesn't change anymore - only changes while playing.

Here are the steps with results:
 1. Execute the application;
 2. Change the volume at run time before playing an audio file;
 3. Play the audio file;
* RESULT: the audio plays at the specified volume, successfully! =)
 4. Press Stop;
 5. Change the volume again;
 6. Press Play again (at this point, the implementation inside the play method should get the new volume information);
* RESULT: the audio plays again at the same volume it played before. Setting the volume doesn't work anymore.

I'm doing my tests on the **Vlc.DotNet.Forms.Samples - CLR 4 - .Net 4.5** project.
The changes I've made to the project were:
 * Added a NumericUpDown control, so that I could change the volume at run time;
 * Associated the ValueChanged event to the NumericUpDown control, so that every time it changes the value, the new value is passed to the VlcControl;
 * Created a Play() function that always gets the last volume value before playing the audio;

My code is below:

    private void numericUpDown1_ValueChanged(object sender, EventArgs e)
    {
        myVlcControl.Audio.Volume = (int)numericUpDown1.Value;
    }

    private void Play()
    {
        myVlcControl.Audio.Volume = (int)numericUpDown1.Value;
        myVlcControl.Play(new FileInfo(FileName));
    }

    private void OnButtonPlayClicked(object sender, EventArgs e)
    {
        Play();
    }

    private void OnButtonStopClicked(object sender, EventArgs e)
    {
        myVlcControl.Stop();
    }

    private void OnButtonPauseClicked(object sender, EventArgs e)
    {
        myVlcControl.Pause();
    }

Any ideas on how to solve this one?

**Comment by jeremyVignelles** [comment]
Seems like a duplicate of #381 (sort of).

If you're still having issues like that, can you try with `--aout=directsound`?

Closing this for now, feel free to reopen if more help is needed.
